frontend/packages/dev-console/src/components/taskruns/list-page/TaskRunsList.tsx

```diff
@@ -1,24 +1,79 @@
 import * as React from 'react';
+import { useTranslation } from 'react-i18next';
 import { Table } from '@console/internal/components/factory';
-import { SortByDirection } from '@patternfly/react-table';
+import { SortByDirection, sortable } from '@patternfly/react-table';
 import { TaskRunModel } from '../../../models';
-import TaskRunsHeader from './TaskRunsHeader';
 import TaskRunsRow from './TaskRunsRow';
+import { tableColumnClasses } from './taskruns-table';
 
 interface TaskRunsListProps {
   customData?: { [key: string]: any };
 }
 
-const TaskRunsList: React.FC<TaskRunsListProps> = (props) => (
-  <Table
-    {...props}
-    aria-label={TaskRunModel.labelPlural}
-    defaultSortField="status.startTime"
-    defaultSortOrder={SortByDirection.desc}
-    Header={TaskRunsHeader(props.customData?.showPipelineColumn)}
-    Row={TaskRunsRow}
-    virtualize
-  />
-);
+const TaskRunsList: React.FC<TaskRunsListProps> = (props) => {
+  const { t } = useTranslation();
+  const TaskRunsHeader = (showPipelineColumn: boolean = true) => () => {
+    return [
+      {
+        title: t('devconsole~Name'),
+        sortField: 'metadata.name',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[0] },
+      },
+      {
+        title: t('devconsole~Namespace'),
+        sortField: 'metadata.namespace',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[1] },
+        id: 'namespace',
+      },
+      {
+        title: t('devconsole~Pipeline'),
+        sortField: 'metadata.labels["tekton.dev/pipeline"]',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[2] },
+      },
+      {
+        title: t('devconsole~Task'),
+        sortField: 'spec.taskRef.name',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[3] },
+      },
+      {
+        title: t('devconsole~Pod'),
+        sortField: 'status.podName',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[4] },
+      },
+      {
+        title: t('devconsole~Status'),
+        sortField: 'status.conditions[0].reason',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[5] },
+      },
+      {
+        title: t('devconsole~Started'),
+        sortField: 'status.startTime',
+        transforms: [sortable],
+        props: { className: tableColumnClasses[6] },
+      },
+      {
+        title: '',
+        props: { className: tableColumnClasses[7] },
+      },
+    ].filter((item) => !(item.title === 'Pipeline' && !showPipelineColumn));
```

Comment:
Since we are localizing `item.title`, this condition will never be true.

Reply:
fixed it.